common/walker/src/report.rs

```diff
@@ -123,9 +126,97 @@ impl SplitScannerError for Result<Report, ScannerError> {
     }
 }
 
+#[derive(Clone, Debug, PartialEq, Eq, serde::Deserialize, serde::Serialize)]
+pub enum ReportType {
+    SBOM(String),
+    VEX(String),
+}
 /// Handle the report
-pub async fn handle_report(report: Report) -> anyhow::Result<()> {
-    // FIXME: this is a very simplistic version of handling the error
-    log::info!("Import report: {report:#?}");
+pub async fn handle_report(report: Report, report_type: ReportType) -> anyhow::Result<()> {
+    let path = env::var_os("REPORT_PATH").unwrap_or_else(|| OsString::from(DEFAULT_REPORT_OUTPUT_PATH));
+    let path = path.to_str().unwrap_or(DEFAULT_REPORT_OUTPUT_PATH).to_string();
+
+    let report_type_string = match report_type {
+        ReportType::SBOM(sbom) => sbom,
+        ReportType::VEX(vex) => vex,
+    };
+
+    let template_content = include_str!("../templates/report.html");
+    let mut tera = Tera::default();
+    tera.add_raw_template("report.html", template_content)?;
+
+    let mut context = Context::new();
+    let current_time = OffsetDateTime::now_utc();
+    context.insert("report", &report);
+    context.insert("type", &report_type_string);
+    context.insert("current_time", &current_time);
+
+    let format = format_description!("[year]-[month]-[day].[hour].[minute].[second]");
```

Comment:
I know that formats are tricky, this one would result into:

```
2024-02-12.12.34.56
```

Which looks a bit hard to read to me. Also the dots might interfere with the meaning of the file extension.

I don't think any of the ISO/RFC date formats help, because either they use colons (which gets tricky when using as a file name) or looks a bit dense (`20240212T123456`).

Doing a quick search I find: https://stackoverflow.com/questions/1248747/what-is-your-favorite-date-and-time-format-in-a-file-name

So what about `2024-02-12_12-34-56`, or `2024-02-12--12-34-56`?